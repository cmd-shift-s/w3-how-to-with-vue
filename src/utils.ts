import MarkdownIt from 'markdown-it'
import hljs from 'highlight.js'
import 'highlight.js/styles/vs.css'

export const md = new MarkdownIt({
  highlight (code, lang) {
    return hljs.highlight(lang, code).value
  }
})
