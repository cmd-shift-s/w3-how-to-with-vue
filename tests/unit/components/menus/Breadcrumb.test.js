import { mount, renderer } from '@/tests/unit/utils'
import Breadcrumb from '@/components/menus/Breadcrumb.vue'

describe('Breadcrumb.vue', () => {
  it('render default', () => {
    const wrap = mount(Breadcrumb, {
      context: {
        props: {
          links: [
            { to: 'javascript:void(0)', title: 'Home' },
            { to: 'javascript:void(0)', title: 'Pictures' },
            { to: 'javascript:void(0)', title: 'Summer 15' },
            'Italy'
          ]
        }
      }
    })

    expect(wrap.classes()).toContain('breadcrumb')
    expect(wrap.findAll('a').length).toEqual(3)
    expect(wrap.findAll('li').length).toEqual(4)

    const $a = wrap.find('a')
    expect($a.element.getAttribute('href')).toEqual('javascript:void(0)')
    expect($a.text()).toEqual('Home')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
