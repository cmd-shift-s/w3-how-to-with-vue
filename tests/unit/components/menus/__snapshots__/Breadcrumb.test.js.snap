// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb.vue render default 1`] = `
<ul class="breadcrumb">
    <li><a href="javascript:void(0)">Home</a></li>
    <li><a href="javascript:void(0)">Pictures</a></li>
    <li><a href="javascript:void(0)">Summer 15</a></li>
    <li>Italy</li>
</ul>
`;
